README.rst

```diff
@@ -1962,6 +1968,344 @@ However, if specifying :code:`Table` objects manually, please make sure to inclu
 for each table and to use only SQL types allowed for the dialect specified in the
 :code:`SQLAlchemySchemaInfo`.
 
+Macro system
+------------
+
+In many real-life situations, the database schema does not fit the user's mental model of the data.
```

Comment:
Two questions: 
- So why not skip ahead to the macro edges section in that case?
- Imagine you are a new user of the compiler, and you're just poking around the docs. If you see a section about macros and think "this is just SQL views reinvented", do you think you'd make it _all the way to the bottom of the entire macro system section_ before you decide this is reinventing the wheel and stop reading?

Reply:
1. You wouldn't skip to the macro edges section in this case because a new reader assumes that the headline section is important to read in order to understand how the components of the macro system work. 
2. That is a fair point. 

I guess that my bigger issue with this section is that it puts together two related, but different subjects: what is the macro system and why did we build it. How about writting it as follows? Give a short introduction of what the macro system is just below the "Macro system" header. Then in a "Motivation" section explain why we built it. That way the user can skim over the motivation section. That way the user the user will get a clear signal that this section is going to talk about the motivation of the system and choose to skip it if he wants to.

Reply:
> a new reader assumes that the headline section is important to read in order to understand how the components of the macro system work.

It is important. It describes what macros fundamentally aim to accomplish: "reshape how you present your data without actually moving or changing it." In my mind, there are two kinds of readers: ones that don't know what macros are meant to be (and should read this section), and ones that understand the idea already and want to know the details of macro edges (and should skip ahead).

@JulianGoetz any advice?

Reply:
My advice was to create a bit of a hybrid of what you both were saying. Rather than separate out problem statement from definition, start with a quick one-sentence definition that establishes the "What" immediately, then moves onto the "Why" and then the "How." The current format as exists on (I believe) Predrag's local is intuitive and effective in my opinion. In short, I think you're both right, we don't want to drop into the Why before establishing the What, but we also don't want to divorce the definition and motivation.